core/arch/mangle_shared.c

```diff
@@ -1640,9 +1640,13 @@ d_r_mangle(dcontext_t *dcontext, instrlist_t *ilist, uint *flags INOUT, bool man
 
 #ifdef AARCH64
         if (instr_is_icache_op(instr) && instr_is_app(instr)) {
-            next_instr = mangle_icache_op(dcontext, ilist, instr, next_instr,
-                                          get_app_instr_xl8(instr) + AARCH64_INSTR_SIZE);
-            continue;
+            instr_t *res =
+                mangle_icache_op(dcontext, ilist, instr, next_instr,
+                                 get_app_instr_xl8(instr) + AARCH64_INSTR_SIZE);
+            if (res) {
+                next_instr = res;
+                continue;
+            }
```

Comment:
This is confusing as it doesn't match the usage of the other mangle routines and is in fact the opposite: better to match their style and return next_instr and then the code here can go back to the simpler way it was before.

Reply:
Delete this and revert to how it was.